flow/layers/opacity_layer.cc

```diff
@@ -108,4 +141,40 @@ void OpacityLayer::Paint(PaintContext& context) const {
   PaintChildren(context);
 }
 
+#if defined(OS_FUCHSIA)
+void OpacityLayer::UpdateScene(SceneUpdateContext& context) {
+  SceneUpdateContext::Transform transform(context,
+     SkMatrix::MakeTrans(offset_.fX, offset_.fY));
+
+  if (kRenderOpacityUsingSystemCompositor) {
+    FML_DCHECK(needs_system_composite());
+
+    // Retained rendering: speedup by reusing a retained entity node if possible.
```

Comment:
The code that handles the retained node seems to be identical to that in the `PhysicalShapeLayer`. Maybe we can distill them into a protected method of `Layer`.